src/components/Playlist/TrackItem/TrackItem.tsx

```diff
@@ -34,6 +44,18 @@ export const TrackItem: React.FC<TrackItem> = ({
 }) => {
     const dispatch = useDispatch();
     const { isActive, isPlaying } = useSelector(checkActiveItemById(id));
+    const existingFavourite = useSelector(getFavouriteById(id));
+    const { showNotification } = useNotification();
+
+    const handleFavourite = () => {
+        // don't add same favourite second time
+        if (existingFavourite) {
+            return;
+        }
+
+        dispatch(addToFavourites.request({ trackId: id }));
+        showNotification("Track was added to favourites", "success");
```

Comment:
Would be better to handle this notification AFTER adding the track to favourites